### ISSUE TITLE
Panic: unprotected underflow while subtracting in block queue

### ISSUE THREAD

**Issue by sisou** [nimiq/core-rs-albatross#551]
I was syncing up to the current devnet, while my node panicked when reaching the tip:

<details>
<summary>Panic log</summary>

```
 2022-01-11 16:58:57.280 INFO  push                 | Ignoring block (#5904768.0, 029326d8ccd97c4d21e62c83e4a5d5d9352ad7d13279a4c99e129ebb4954fe5e)- we already know a later macro block (we're at 5904782)
 2022-01-11 16:58:57.727 INFO  network              | Connection established with peer 12D3KooWMcaDYVPpDUnXFVFHHSC4fZBaZHqgsVYPrMqyVmAuP22n, Dialer { address: "/ip4/3.144.222.84/tcp/8443/ws/p2p/12D3KooWMcaDYVPpDUnXFVFHHSC4fZBaZHqgsVYPrMqyVmAuP22n" }, connections established: 1 
 2022-01-11 16:58:57.729 INFO  network              | Connection closed with peer 12D3KooWMcaDYVPpDUnXFVFHHSC4fZBaZHqgsVYPrMqyVmAuP22n, Dialer { address: "/ip4/3.144.222.84/tcp/8443/ws/p2p/12D3KooWMcaDYVPpDUnXFVFHHSC4fZBaZHqgsVYPrMqyVmAuP22n" }, connections established: 0 
 2022-01-11 16:58:57.729 INFO  network              | Connection closed because: IO(Custom { kind: Other, error: Closed }) 
 2022-01-11 16:58:59.805 ERROR panic                | thread 'tokio-runtime-worker' panicked at 'attempt to subtract with overflow': /home/soeren/nimiq/core-rs-albatross/consensus/src/sync/block_queue.rs:212
   0: log_panics::init::{{closure}}
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/log-panics-2.0.0/src/lib.rs:52:25
   1: std::panicking::rust_panic_with_hook
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/panicking.rs:610:17
   2: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/panicking.rs:500:13
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/sys_common/backtrace.rs:139:18
   4: rust_begin_unwind
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/panicking.rs:498:5
   5: core::panicking::panic_fmt
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/core/src/panicking.rs:107:14
   6: core::panicking::panic
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/core/src/panicking.rs:48:5
   7: nimiq_consensus::sync::block_queue::Inner<N>::on_block_announced
             at consensus/src/sync/block_queue.rs:212:21
   8: nimiq_consensus::sync::block_queue::BlockQueue<N,TReq>::push_block
             at consensus/src/sync/block_queue.rs:528:9
   9: <nimiq_consensus::consensus::Consensus<N> as core::future::future::Future>::poll
             at consensus/src/consensus/mod.rs:331:21
  10: tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/core.rs:161:17
  11: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/loom/std/unsafe_cell.rs:14:9
  12: tokio::runtime::task::core::CoreStage<T>::poll
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/core.rs:151:13
  13: tokio::runtime::task::harness::poll_future::{{closure}}
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:461:19
  14: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/core/src/panic/unwind_safe.rs:271:9
  15: std::panicking::try::do_call
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/panicking.rs:406:40
  16: __rust_try
  17: std::panicking::try
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/panicking.rs:370:19
  18: std::panic::catch_unwind
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/panic.rs:133:14
  19: tokio::runtime::task::harness::poll_future
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:449:18
  20: tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:98:27
  21: tokio::runtime::task::harness::Harness<T,S>::poll
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:53:15
  22: tokio::runtime::task::raw::poll
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/raw.rs:113:5
  23: tokio::runtime::task::raw::RawTask::poll
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/raw.rs:70:18
  24: tokio::runtime::task::LocalNotified<S>::run
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/mod.rs:343:9
  25: tokio::runtime::thread_pool::worker::Context::run_task::{{closure}}
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/thread_pool/worker.rs:443:21
  26: tokio::coop::with_budget::{{closure}}
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/coop.rs:102:9
  27: std::thread::local::LocalKey<T>::try_with
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/thread/local.rs:399:16
  28: std::thread::local::LocalKey<T>::with
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/thread/local.rs:375:9
  29: tokio::coop::with_budget
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/coop.rs:95:5
      tokio::coop::budget
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/coop.rs:72:5
      tokio::runtime::thread_pool::worker::Context::run_task
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/thread_pool/worker.rs:419:9
  30: tokio::runtime::thread_pool::worker::Context::run
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/thread_pool/worker.rs:386:24
  31: tokio::runtime::thread_pool::worker::run::{{closure}}
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/thread_pool/worker.rs:371:17
  32: tokio::macros::scoped_tls::ScopedKey<T>::set
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/macros/scoped_tls.rs:61:9
  33: tokio::runtime::thread_pool::worker::run
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/thread_pool/worker.rs:368:5
  34: tokio::runtime::thread_pool::worker::Launch::launch::{{closure}}
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/thread_pool/worker.rs:347:45
  35: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/blocking/task.rs:42:21
  36: tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/core.rs:161:17
  37: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/loom/std/unsafe_cell.rs:14:9
  38: tokio::runtime::task::core::CoreStage<T>::poll
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/core.rs:151:13
  39: tokio::runtime::task::harness::poll_future::{{closure}}
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:461:19
  40: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/core/src/panic/unwind_safe.rs:271:9
  41: std::panicking::try::do_call
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/panicking.rs:406:40
  42: __rust_try
  43: std::panicking::try
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/panicking.rs:370:19
  44: std::panic::catch_unwind
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/panic.rs:133:14
  45: tokio::runtime::task::harness::poll_future
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:449:18
  46: tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:98:27
  47: tokio::runtime::task::harness::Harness<T,S>::poll
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:53:15
  48: tokio::runtime::task::raw::poll
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/raw.rs:113:5
  49: tokio::runtime::task::raw::RawTask::poll
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/raw.rs:70:18
  50: tokio::runtime::task::UnownedTask<S>::run
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/mod.rs:379:9
  51: tokio::runtime::blocking::pool::Inner::run
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/blocking/pool.rs:264:17
  52: tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}
             at /home/soeren/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/blocking/pool.rs:244:17
  53: std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/sys_common/backtrace.rs:123:18
  54: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/thread/mod.rs:484:17
  55: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/core/src/panic/unwind_safe.rs:271:9
  56: std::panicking::try::do_call
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/panicking.rs:406:40
  57: __rust_try
  58: std::panicking::try
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/panicking.rs:370:19
  59: std::panic::catch_unwind
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/panic.rs:133:14
  60: std::thread::Builder::spawn_unchecked::{{closure}}
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/thread/mod.rs:483:30
  61: core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/core/src/ops/function.rs:227:5
  62: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/alloc/src/boxed.rs:1694:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/alloc/src/boxed.rs:1694:9
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/6db0a0e9a4a2f55b1a85954e114ada0b45c32e45/library/std/src/sys/unix/thread.rs:108:17
  63: start_thread
  64: __clone3

```
</details>

**Comment by styppo** [comment]
This panic is already fixed in cecda6b2db0bd1983e5876babde579fcc9d14d39, please try again with the latest `albatross` branch.

**Comment by sisou** [comment]
It does fix it. I was testing at 6dd8da85 to be able to connect to devnet (there are changes between that and cecda6b that prevent connection to the seed node). But cherry picking it onto 6dd8da85, it does not appear anymore :+1:.
